code/game/machinery/cell_charger.dm

```diff
@@ -144,6 +146,9 @@
 	else
 		use_power(transfer_rate*transfer_rate_coeff) //Snatch some power
 		charging.give(transfer_rate*transfer_rate_coeff*(transfer_efficiency+transfer_efficiency_bonus)) //Inefficiency (Joule effect + other shenanigans)
```

Comment:
```
use_power(transfer_rate * transfer_rate_coeff * emagged ? 10 : 1)  //Drain all the power if emagged
if(charging.give(transfer_rate*transfer_rate_coeff * (transfer_efficiency+transfer_efficiency_bonus) * emagged ? 0.25 : 1) //Inefficiency (Joule effect + other shenanigans)  //Lose most of it if emagged
 if(has_beeped) //It's charging again
  has_beeped = FALSE
```
give() will return the charged ammount, so you can use it to check if your cell is charging to cover that edge case that i commented about. You can even move the use_power() to under that if to prevent power from being consumed if the cell isn't charging, too.